optimizely/event/processor_test.go

```diff
@@ -30,27 +30,103 @@ func TestDefaultEventProcessor_ProcessImpression(t *testing.T) {
 
 }
 
-func TestDefaultEventProcessor_ProcessImpressions(t *testing.T) {
-	processor := NewEventProcessor(100, 100)
+type FakeDispatcher struct {
```

Comment:
nit: I've been calling everything mock*. So this can be MockDispatcher. 

Also, I've been using the `mock` function from the `testify` package to assert that function calls are made.